tests/apollo/util/skvbc.py

```diff
@@ -482,16 +519,25 @@ async def run_concurrent_conflict_ops(self, num_ops, write_weight=.70):
         max_size = len(self.keys) // 2
         return await self.send_concurrent_ops(num_ops, max_concurrency, max_size, write_weight, create_conflicts=True)
 
-    async def send_concurrent_ops(self, num_ops, max_concurrency, max_size, write_weight, create_conflicts=False):
+    async def send_concurrent_ops(self, num_ops, max_concurrency, max_size, write_weight, create_conflicts=False,write_run_duration=None):
```

Comment:
I think it should be more general,  the way the function is called is unclear like that. Two parameters are used for the limit and the caller must-knows of the exact internal implementation. I would also add a comment to explain. Here is a pseudo-code example (haven't checked it) in a clearer way:

```
from enum import Enum # should be added at the top of the file
    class ExitPolicy(Enum):
        COUNT = 0
        TIME = 1

    async def send_concurrent_ops(self, max_concurrency, max_size, write_weight, exit_factor, create_conflicts=False, exit_policy=ExitPolicy.COUNT):
        ''' exit_factor should be number of seconds to run if exit_policy is TIME and number of times to run if exit_policy is COUNT '''
        max_read_set_size = max_size
        if self.tracker is not None:
            max_read_set_size = 0 if self.tracker.no_conflicts else max_size
        sent = 0
        write_count = 0
        read_count = 0
        if exit_policy is ExitPolicy.COUNT:
            total_run_duration = time.time()
            total_run_duration += exit_factor
        clients = self.bft_network.random_clients(max_concurrency)

        def exit_predicate():
            if exit_policy is ExitPolicy.COUNT:
                return sent < exit_factor
            if exit_policy is ExitPolicy.TIME:
                return time.time() < total_run_duration
            assert false, "Unknown exit policy"
...
...
...
```